### ISSUE TITLE
P4-A: Train fast rubric scorer via distillation

### ISSUE THREAD

**Issue by jpequegn** [jpequegn/rubric-gates#27]
## Description — Phase A: Scorer Distillation

**Goal:** Replace expensive LLM-as-judge calls (P1 documentation + testability scorers) with a fast local model that predicts rubric scores.

**This is rubric engineering in the narrow sense:** the rubric scores from P1 become training labels, and we train a model to reproduce them.

## Why This Matters

P1's documentation and testability scorers use Claude API calls (LLM-as-judge). Each scoring event costs money and adds latency. By training a small local model on accumulated scores, you get:
- **10-100x faster scoring** (local inference vs API call)
- **Zero marginal cost** per score after training
- **Offline capability** — scoring works without internet
- **Consistent scores** — no API variability

### Data
- **Source:** P1 scorecard storage — accumulated (code, dimension_scores) pairs
- **Minimum:** 500+ real examples, augmented with synthetic data to reach 2000+
- **Format:** `(code_text, dimension, score)` triples

### Model
- **Base:** Small code model — CodeGemma-2B or DeepSeek-Coder-1.3B (fits on consumer GPU with 4-bit quantization)
- **Fine-tuning:** LoRA (parameter-efficient, trains in hours not days)
- **Output:** Regression head predicting 0.0-1.0 score per dimension

### Training
- **Method:** Supervised fine-tuning (SFT) on score prediction
- **Loss:** MSE between predicted and actual rubric scores
- **Validation:** Hold out 20% of real data, measure prediction accuracy
- **Target:** < 0.1 MAE (mean absolute error) vs LLM judge scores

### Architecture Options

**Option A: Multi-head regression (recommended for v1)**
```
[code] → [encoder] → [shared features] → [correctness_head] → 0.0-1.0
                                        → [security_head]     → 0.0-1.0
                                        → [maintainability_head] → 0.0-1.0
                                        → [documentation_head] → 0.0-1.0
                                        → [testability_head]   → 0.0-1.0
```

**Option B: Text-to-score (simpler)**
```
Input: "Score this code for documentation:\n{code}"
Output: "0.72"
```

## Interface

```python
class DistilledScorer:
    """Fast local model that predicts rubric scores."""
    
    def __init__(self, model_path: str):
        """Load trained model from checkpoint."""
    
    def score(self, code: str, dimension: Dimension) -> float:
        """Predict score for a single dimension. Returns 0.0-1.0."""
    
    def score_all(self, code: str) -> dict[Dimension, float]:
        """Predict all dimension scores at once."""

class ScorerDistillationTrainer(RubricTrainer):
    """Train a fast scorer from accumulated rubric data."""
    
    def prepare_data(self, storage: StorageBackend) -> Dataset:
        """Load and prepare training data from P1 scores."""
    
    def train(self, epochs: int = 10, lr: float = 1e-4) -> TrainingResult:
        """Fine-tune the model on score prediction."""
    
    def evaluate(self, test_data: Dataset) -> EvalResult:
        """Evaluate prediction accuracy vs LLM judge."""
    
    def export(self, path: str) -> None:
        """Export trained model for use as DistilledScorer."""
```

## Integration with P1

Once trained, the distilled scorer can replace LLM-judge calls:

```yaml
# .rubric-gates.yaml
scorecard:
  dimensions:
    documentation:
      weight: 0.15
      enabled: true
      method: distilled  # Use local model instead of LLM judge
      model_path: ./models/scorer-v1/
    testability:
      weight: 0.15
      enabled: true
      method: distilled
      model_path: ./models/scorer-v1/
```

## Calibration

The distilled model must be calibrated against the original LLM judge:
- **Platt scaling** — fit a logistic regression on held-out data to calibrate probabilities
- **Temperature scaling** — single parameter to sharpen/soften predictions
- Measure Expected Calibration Error (ECE) — target < 0.05

## Acceptance Criteria

- [ ] Trainer in `training/distillation.py`
- [ ] Scorer in `training/distilled_scorer.py`
- [ ] Training works on local GPU (tested with 8GB+ VRAM)
- [ ] Prediction MAE < 0.1 vs LLM judge on held-out data
- [ ] Inference latency < 100ms per file (vs seconds for API)
- [ ] Config option to swap LLM judge for distilled scorer
- [ ] Calibration metrics computed and logged
- [ ] Tests with mock model (CI) and real model (local GPU)

## Dependencies

- #26 (training infrastructure)
- #7, #8 (documentation + testability scorers — provide training labels)
- #3 (storage — training data source)
- #2 (config — method switching)
